PocketKit/Sources/Sync/RemoteMapping/SavedItem+RemoteMapping.swift

```diff
@@ -34,7 +34,21 @@ extension SavedItem {
             let itemParts = remote.item.fragments.itemParts else {
             return
         }
-
+        
+        if let tags = tags {
+            removeFromTags(tags)
+        }
+        
+        remote.tags?.compactMap { $0 }.forEach({ remoteTag in
+            let fetchRequest = Requests.fetchTag(byName: remoteTag.name)
```

Comment:
If we extract this logic out of from the method in `Source` (see above comment), we could reuse it here.

Reply:
okay! from the current ticket I have, I realize I have to rethink this logic since it doesn't delete old tags (not assigned to save items) from core data. I'll postpone that work in a different PR though

Reply:
I realize that this is an extension on saved item and doesn't have access to `source`.